src/runtime/HalideRuntimeOpenGLCompute.h

```diff
@@ -0,0 +1,50 @@
+#ifndef HALIDE_HALIDERUNTIMEOPENGLCOMPUTE_H
+#define HALIDE_HALIDERUNTIMEOPENGLCOMPUTE_H
+
+#include "HalideRuntime.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** \file
+ *  Routines specific to the Halide OpenGL Compute runtime.
+ */
+
+extern const struct halide_device_interface *halide_openglcompute_device_interface();
+
+/** These are forward declared here to allow clients to override the
+ *  Halide Glsl runtime. Do not call them. */
+// @{
+extern int halide_openglcompute_initialize_kernels(void *user_context, void **state_ptr,
+                                            const char *src, int size);
+
+extern int halide_openglcompute_run(void *user_context,
+                             void *state_ptr,
+                             const char *entry_name,
+                             int blocksX, int blocksY, int blocksZ,
+                             int threadsX, int threadsY, int threadsZ,
+                             int shared_mem_bytes,
+                             size_t arg_sizes[],
+                             void *args[],
+                             int8_t is_buffer[],
+                             int num_attributes,
+                             float* vertex_buffer,
+                             int num_coords_dim0,
+                             int num_coords_dim1);
+// @}
+
+/** This functions MUST be provided by the host environment to retrieve pointers
```

Comment:
So it's not true that it must be provided by the host environment. It only
needs to be provided on platforms other than linux, osx and android in the
opengl case, and platforms other than android in the openglcompute case.

Maybe something like "You may have to implement this yourself. Halide only
provides implementations for some platforms."

On Fri, Jul 24, 2015 at 1:18 PM, Alexander Aprelev <notifications@github.com
> wrote:

> In src/runtime/HalideRuntimeOpenGLCompute.h
> <https://github.com/halide/Halide/pull/864#discussion_r35461720>:
>
> > +extern int halide_openglcompute_run(void *user_context,
> > +                             void *state_ptr,
> > +                             const char *entry_name,
> > +                             int blocksX, int blocksY, int blocksZ,
> > +                             int threadsX, int threadsY, int threadsZ,
> > +                             int shared_mem_bytes,
> > +                             size_t arg_sizes[],
> > +                             void *args[],
> > +                             int8_t is_buffer[],
> > +                             int num_attributes,
> > +                             float* vertex_buffer,
> > +                             int num_coords_dim0,
> > +                             int num_coords_dim1);
> > +// @}
> > +
> > +/** This functions MUST be provided by the host environment to retrieve pointers
>
> I think comment(inherited from HalideRuntimeOpenGL.h) makes sense. It's
> just happens that there are implementations of this function for Linux, OSX
> and Android platforms. If you are on some other platform, you have to
> provide your own implementation.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/864/files#r35461720>.
>